src/ripple/app/main/Application.cpp

```diff
@@ -171,10 +175,49 @@ class AppFamily : public shamap::Family
     void
     missing_node (std::uint32_t seq) override
     {
-        uint256 const hash = getApp().getLedgerMaster ().getHashBySeq (seq);
-        if (hash.isZero())
+        WriteLog (lsERROR, Ledger) << "Missing node in " << seq;
+
+        // prevent recursive invocation
+        std::unique_lock <std::mutex> lock (maxSeqLock);
```

Comment:
I'm trying to understand the `prevent recursive invocation` comment for this lock. I can see the lock protects `maxSeq`, it is correctly unlocked before calls that might invoke the missing node handler (which I assume can recursively call into `missing_node`, but haven't checked). All looks well.

But the lock isn't preventing a recursive call. Reword/remove the comment?